### DISCUSSION TITLE
Smile Coin - Streak Data

### DISCUSSION THREAD

**Discussion by marucat27** [asastats/channel#322] [General]
I was curious if it would be possible to show Smile Coin users game streak data via ASA Stats.  If a smile user plays a parter game such as grin gaming once a day, they build a streak.  I believe a 7 day streak provide 25% apy if staked through their dapp, and 125% apy if have a 30 day game streak.  They also keep track of when you have to play next to keep up that streak like a countdown timer.  There is public api for this per one of the team members.  Any thoughts?  Streak site:  https://www.smileodds.org/

Discussion between Maru and Massimo from Smile Team via Telegram:

Maru: Hey Massimo!
I’m curious is the smile streak info on the blockchain or is this from smiles website.  I’m wondering if it’s something that can be shown on the ASA Stats webpage? Like streaks?  I know they can read or pull data if the info is on chain.  Just curious.  Or if there is APIs to pull this info and if those are available.  If not, no worries.  Thanks!

Massimo: It's all on the blockchain, which is why I can get the same (often more exact LOL) data on SmileOdds. The data, however, must be processed with some logic, the blockchain contains just the transaction, identified by text in the notes, and the timestamp. So to get a data like streak you have to use a programming language (more or less all are suitable, although Python and Javascript simplify things a bit)

Maru: Is this something that could be provided via api?  I use Asa stats a lot like many and often provide suggestions to them in their discord to add new features.  Adding something like streak data would be nice for smile users.  Of course your site is excellent as well.

Massimo: the streak has a public API https://api.smilecoin.us/api/v1/streak/AAAAAAAAA

where AAAAAAAAA is the address

**Comment by ipaleka** [comment]
After a quick research, it looks like that app rewards as a consequence of some other actions. Basically, you have xy SMILE in your account and after some time you get xy + z SMILE in your account. From our perspective we do track those accounts as in every moment we correctly track user SMILE in their account.

For example, as a rule of thumb, we don't implement a reward count for user staking/farming. There are a couple of reasons behind that policy. First of all, it brings a double or triple development effort. But the primary reason is that we don't want to steal page hits from those providers, we are an ecosystem project and we want that our users visit other projects in order to accomplish things those projects bring. For example, we don't show the "Next payout" value from https://app.smilecoin.us/rewards/ .

That being said, this Smile Odds isn't something we should implement in any form. What we **have to** implement is the following:

> if staked through their dapp

Besides that https://app.smilecoin.us/rewards/ dApp, is there some other dApp where people stake their SMILE and after that action their SMILE don't show up in ASA Stats?

**Reply by marucat27** [reply]
No I believe ASA Stats already captures the correct amount of smile staked or held in wallet.  On the smile dapp users only commit a certain amount to be staked similar to Algorand governance. 👍

**Reply by ipaleka** [reply]
> No I believe ASA Stats already captures the correct amount of smile staked or held in wallet. On the smile dapp users only commit a certain amount to be staked similar to Algorand governance. +1

Yes, they changed their staking dApp a few months ago, from locking in an escrow to the current implementation. No staking in those games also, right?

**Reply by marucat27** [reply]
> > No I believe ASA Stats already captures the correct amount of smile staked or held in wallet. On the smile dapp users only commit a certain amount to be staked similar to Algorand governance. +1
> 
> 
> 
> Yes, they changed their staking dApp a few months ago, from locking in an escrow to the current implementation. No staking in those games also, right?

Correct, no staking in the games.

**Comment by ipaleka** [comment]
I'm using this opportunity to emphasize some other things.

First of all, @marucat27 this is a brilliant example how things should be done in ASA Stats. You're a kind of new community member, but you've manage to get grasp on things perfectly.

Everything is subject for discussion in ASA Stats, while an issue is something that has to be resolved. You have correctly opened this discussion and even made your own research prior to that.

Next, don't be disappointed that we're not going to implement this in our core engine. The thing is that everybody (maybe even we by ourselves) can implement what you've requested inside a dedicated user widget! Once someone takes that step then they will be refered exactly to this thread/discussion. And they can find a result of your research here (API for example).

Whenever somebody point at this app, we should redirect such a discussion here, even by copying and pasting conversation here from some other place.

Thank you Maru! :)
